sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpMessageExtensions.cs

```diff
@@ -124,5 +123,17 @@ public static DateTimeOffset GetScheduledEnqueueTime(this AmqpAnnotatedMessage m
             }
             return default;
         }
+
+        public static BinaryData GetBody(this AmqpAnnotatedMessage message)
+        {
+            if (message.Body is AmqpDataBody dataBody)
+            {
+                return dataBody.Data.ConvertAndFlattenData();
+            }
+            else
```

Comment:
nit:  no need for `else` here;  you'll have already returned if you met the `if` condition.